core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -104,13 +90,70 @@ object RequestChannel extends Logging {
       }
     }
 
-    trace(s"Processor $processor received request: ${requestDesc(true)}")
+    trace(s"Processor $processor received request: ${summary.getDescription(true)}")
+
+    val summary: RequestSummary = new RequestSummary(
+      processor,
+      header,
+      metrics,
+      context,
+      session,
+      startTimeNanos,
+      description = s"${context.header} -- ${bodyAndSize.request.toString(false)}",
+      detailedDescription = s"${context.header} -- ${bodyAndSize.request.toString(true)}",
+      sizeInBytes = bodyAndSize.size,
+      isFromFollower = bodyAndSize.request match {
+        case r:FetchRequest => Some(r.isFromFollower)
+        case _ => None
+      }
+    )
+
+    def releaseBuffer(): Unit = {
+      if (buffer != null) {
+        memoryPool.release(buffer)
+        buffer = null
+      }
+    }
+
+    override def toString = s"Request(processor=$processor, " +
+      s"connectionId=${context.connectionId}, " +
+      s"session=$session, " +
+      s"listenerName=${context.listenerName}, " +
+      s"securityProtocol=${context.securityProtocol}, " +
+      s"buffer=$buffer)"
+  }
+
+  /** Summary of Request without the parsed request **/
+  class RequestSummary(
+    val processor: Int,
+    val header: RequestHeader,
+    val metrics: Metrics,
+    val context: RequestContext,
+    session: Session,
+    startTimeNanos: Long,
+    description: String,
+    detailedDescription: String,
+    val sizeInBytes: Int,
+    isFromFollower: Option[Boolean]
```

Comment:
Maybe use just a regular Boolean here? I'm not sure what is achieved by wrapping it in an optional. Above you could do something like:

```scala
     isFromFollower = bodyAndSize.request match {
        case r:FetchRequest =>r.isFromFollower
        case _ => false
      }
```

Reply:
The reason I used an `Option` for `isFromFollower` is to indicate a `None` when the request isn't actually a `FetchRequest`, but you're right it's a bit superfluous since there's only one condition checking if it's `true`.